### PR TITLE
Feature/notification center crw 140

### PR DESCRIPTION
**Pull request by AbdallaEmadEldinAbdalla**
[CRW-140](https://techxarya.atlassian.net/browse/CRW-140)
[CRW-196](https://techxarya.atlassian.net/browse/CRW-196)
[CRW-141](https://techxarya.atlassian.net/browse/CRW-141)

[CRW-140]: https://techxarya.atlassian.net/browse/CRW-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRW-196]: https://techxarya.atlassian.net/browse/CRW-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRW-141]: https://techxarya.atlassian.net/browse/CRW-141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/notifications/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const { notifications } = useSelector(({ notifications }: any) => notifications);
+
+    useEffect(() => {
+        // setIsLoading(true);
+        // getNotifications(0, 100, 'asc').then(() => {
+        //     setIsLoading(false);
+        // }).catch(() => {
+        //     setIsLoading(false);
+        // });
+    }, []);
```
Comment (almujtabamohammed) [inline src/pages/notifications/index.tsx:30]: Remove

File: src/pages/notifications/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    <LoadingSpinner />
+                    : notifications.length <= 0 ?
+                        <Col className="items-center mt-14 gap-6">
+                            <EmptyIcon />
+                            <p className="font-bold text-lg text-grey-1">{t('emptyList')}</p>
+                        </Col>
+                        : notifications.map((notification: NotificationType, index: number, arr: NotificationType[]) => {
+                            const isLastItem = index === arr.length - 1;
+                            return (
+                                <NotificationCard notification={notification} type="fullscreen" isLastItem={isLastItem} />
```
Comment (almujtabamohammed) [inline src/pages/notifications/index.tsx:47]: Add key for list items

File: src/services/redux/notificationsSlice.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    notifications: [],
+    hasNewNotifications: false,
+};
+
+export const notificationsSlice = createSlice({
+    name: "notifications",
+    initialState,
+    reducers: {
+        setNotifications: (state, action: { payload: NotificationType[] }) => {
+            console.log({ xxx: action.payload });
```
Comment (almujtabamohammed) [inline src/services/redux/notificationsSlice.tsx]: Remove

File: src/services/redux/notificationsSlice.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { NotificationType } from "../../types/notifications";
+
+interface initialStateType {
+    notifications: NotificationType[];
+    hasNewNotifications: boolean;
+}
+// TODO: remove this dummy data.
```
Comment (almujtabamohammed) [inline src/services/redux/notificationsSlice.tsx:8]: Remove TODO

File: src/utils/notifications.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        case "STOP_LOSS_EXECUTED":
+        case "TRAILIING_EXECUTED":
+            return <ExchangeImage providerId={id} width={26} height={26} />
+        case "KEYS_EXPIRATION_SOON":
+        case "KEYS_EXPIRED":
+            return <Cog6ToothIcon className="stroke-current w-7 h-7" />;
+        case "SMART_ALLOCATION_REBALANCED":
+        case "EXIT_STRATEGY_EXECUTED":
+            return <PortfolioDefault className="stroke-white w-7 h-7" color="#0E1421" fill="white" />;
+    }
```
Comment (almujtabamohammed) [inline src/utils/notifications.tsx:38]: No default icon?

File: src/pages/notifications/index.tsx
Comment (almujtabamohammed) [inline src/pages/notifications/index.tsx:1]: - Mark notifications as seen
- Add pagination and see more

File: src/components/layout/Nav/Nav.tsx [hunk 6]
```diff
@@ -139,15 +165,75 @@ const Nav = () => {
...
+
+                {notifications.length <= 0 ?
+                  <Col className="items-center mt-14 gap-6">
+                    <EmptyIcon />
+                    <p className="font-bold text-lg text-grey-1">{t('notification:emptyList')}</p>
+                  </Col>
+                  :
+                  notifications.map((notification: NotificationType) => {
+                    return (
+                      <NotificationCard notification={notification} type="dropdown" />
```
Comment (almujtabamohammed) [inline src/components/layout/Nav/Nav.tsx:222]: Add key

**Review by almujtabamohammed — CHANGES_REQUESTED** [review]

File: src/pages/notifications/index.tsx (hunk 1 above)
Comment (AbdallaEmadEldinAbdalla) [reply src/pages/notifications/index.tsx:30]: I will handle the pagination here thats why i left the api call @almujtabamohammed

File: src/utils/notifications.tsx (hunk 5 above)
Comment (AbdallaEmadEldinAbdalla) [reply src/utils/notifications.tsx:38]: it's not exists on the UI.

**Pull request merged**